_studio/mfx_lib/decode/mpeg2/src/mfx_mpeg2_decode.cpp

```diff
@@ -1884,7 +1884,7 @@ mfxStatus VideoDECODEMPEG2InternalBase::GetVideoParam(mfxVideoParam *par)
         m_implUmc->GetSignalInfoInformation(signal_info);
     }
 
-    memcpy_s(par->reserved,sizeof(m_vPar.reserved),m_vPar.reserved,sizeof(m_vPar.reserved));
+    std::copy(m_vPar.reserved, m_vPar.reserved + sizeof(m_vPar.reserved), par->reserved);
```

Comment:
please, avoid low-level memory management whenever it possible (sizeof is part of it):
```
std::copy(std::begin(m_vPar.reserved), std::end(m_vPar.reserved), par->reserved);
```